src/nvme/nbft.c

```diff
@@ -0,0 +1,947 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/*
+ * This file is part of libnvme.
+ * Copyright (c) 2021-2022, Dell Inc. or its subsidiaries.  All Rights Reserved.
+ *
+ * Authors: Stuart Hayes <Stuart_Hayes@Dell.com>
+ *
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <arpa/inet.h>
+
+#include "private.h"
+#include "nbft.h"
+#include "log.h"
+
+/*
+ *  ACPI NBFT table structures (spec v0.65)
+ */
+
+#define NBFT_ACPI_SIG		"NBFT"
+
+enum nbft_raw_descriptor_type {
+	NBFT_HEADER,
+	NBFT_CONTROL,
+	NBFT_HOST,
+	NBFT_HFI,
+	NBFT_SSNS,
+	NBFT_SECURITY,
+	NBFT_DISCOVERY,
+	NBFT_HFI_TRANSPORT,
+	RESERVED_8,
+	NBFT_SSNS_EXTENDED_INFO,
+};
+
+typedef struct __attribute__((__packed__)) heap_obj_s {
+	__u32 offset;
+	__u16 length;
+} heap_obj;
+
+/*
+ * HEADER (Figure 8)
```

Comment:
Add a self containing documentation and only reference for additional information to external resources. I just mention it once here. There are more of these comments.

Reply:
Fixed.